Sources/NIOHTTP1/ByteCollectionUtils.swift

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+fileprivate let defaultWhitespaces = [" ", "\t"].map({$0.utf8.first!})
+
+extension ByteBufferView {
+    internal func trim(limitingElements: [UInt8]) -> ByteBufferView {
+        let lastNonWhitespaceIndex = self.lastIndex { !limitingElements.contains($0) }
+        let firstNonWhitespaceIndex = self.firstIndex { !limitingElements.contains($0) }
```

Comment:
Ah, good spot that these are actually optional. If they're optional, let's not force-unwrap these: these should be `guard let` instead. These only end up being nil if nothing matches the predicate. If nothing matches the predicate, then the buffer is entirely trimmed elements, so we want to return the empty `ByteBufferView`:

```swift
guard let lastNonWhitespaceIndex = self.lastIndex({ !limitingElements.contains($0) }),
      let firstNonWhitespaceIndex = self.firstIndex({ !limitingElements.contains($0) }) else {
    // This buffer is entirely trimmed elements, so trim it to nothing.
    return self[self.startIndex..<self.startIndex]
}
```

Then we don't need the force-unwraps.